cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedElement.qll

```diff
@@ -442,11 +442,16 @@ newtype TTranslatedElement =
   // The declaration/initialization part of a `ConditionDeclExpr`
   TTranslatedConditionDecl(ConditionDeclExpr expr) { not ignoreExpr(expr) } or
   // The side effects of a `Call`
-  TTranslatedSideEffects(Call expr) {
-    exists(TTranslatedArgumentSideEffect(expr, _, _, _)) or
-    expr instanceof ConstructorCall or
-    expr.getTarget() instanceof AllocationFunction
-  } or // A precise side effect of an argument to a `Call`
+  TTranslatedCallSideEffects(Call expr) {
+    not expr instanceof AllocationExpr and
```

Comment:
Please add a comment explaining this line. IIUC, the explanation is that
1. We can omit the `TTranslatedCallSideEffects` translated element because it doesn't have instructions but just serves as a container for the specific side effects in its children, and those children get moved to the `TTranslatedAllocationSideEffects` translated element.
2. No call can be both an `AllocationExpr` and a `ConstructorCall`.

Reply:
This sounds like the right explanation to me - I will add that!